src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -767,58 +831,203 @@ public IEnumerable<string> GetCollections(Transaction transaction)
+                var tvr = read.Result.Reader;
+                using var revision = TableValueToRevision(context, ref tvr);
+
+                if (minimumTimeToKeep.HasValue &&
+                    _database.Time.GetUtcNow() - revision.LastModified <= minimumTimeToKeep.Value)
+                    break;
+
+                var collection = CollectionName.GetCollectionName(revision.Data);
+
+                revisionsToRemove.Add((revision.ChangeVector, revision.Etag, revision.Flags, collection));
+                deletedRevisionsCount++;
+            }
+
+            var maxEtagDeleted = DeleteRevisionsInternal(context, table, collectionName, changeVector, lastModifiedTicks, revisionsToRemove);
+
+            _database.DocumentsStorage.EnsureLastEtagIsPersisted(context, maxEtagDeleted);
+            return deletedRevisionsCount;
+        }
+
+        private long DeleteRevisionsInternal(DocumentsOperationContext context, Table table, CollectionName collectionName,
+            string changeVector, long lastModifiedTicks, 
+            List<(string ChangeVector, long Etag, DocumentFlags Flags, string CollectionName)> revisionsToRemove)
+        {
+            var writeTables = new Dictionary<string, Table>();
+            long maxEtagDeleted = 0;
 
-                foreach (var read in table.SeekForwardFrom(RevisionsSchema.Indexes[IdAndEtagSlice], prefixSlice, skip: 0, startsWith: true))
+            foreach (var revisionInfo in revisionsToRemove)
+            {
+                using (Slice.From(context.Allocator, revisionInfo.ChangeVector, out var keySlice))
                 {
-                    if (numberOfRevisionsToDelete <= deletedRevisionsCount)
-                        break;
+                    CreateTombstone(context, keySlice, revisionInfo.Etag, collectionName, changeVector, lastModifiedTicks);
 
-                    var tvr = read.Result.Reader;
-                    using (var revision = TableValueToRevision(context, ref tvr))
+                    maxEtagDeleted = Math.Max(maxEtagDeleted, revisionInfo.Etag);
+                    if ((revisionInfo.Flags & DocumentFlags.HasAttachments) == DocumentFlags.HasAttachments)
                     {
-                        if (minimumTimeToKeep.HasValue &&
-                            _database.Time.GetUtcNow() - revision.LastModified <= minimumTimeToKeep.Value)
-                            return deletedRevisionsCount;
+                        if (table.SeekOnePrimaryKeyPrefix(keySlice, out TableValueReader tvr))
+                        {
+                            using (var revision = TableValueToRevision(context, ref tvr))
+                                _documentsStorage.AttachmentsStorage.DeleteRevisionAttachments(context, revision, changeVector, lastModifiedTicks);
+                        }
+                        else
+                        {
+                            throw new VoronErrorException($"Failed to remove revision with change vector {revisionInfo.ChangeVector}");
```

Comment:
can this happened? did it ever happened to you?

Reply:
I don't think so, putted it there just in case